maps-app-ios/UI/NorthArrowView.swift

```diff
@@ -56,8 +56,14 @@ public class NorthArrowView: RoundedImageView {
     }
     
     func setVisibilityFromMapView(animate:Bool = false) {
+        DispatchQueue.main.async {
```

Comment:
Are you intending on increasing the retain count of self here? I don't see an issue with it with main queue async code, i'm curious because you capture self weakly on line 112 for other UI updates.

Reply:
Yeah, probably not. I'll fix.